### PR TITLE
Task/51 get budget

### LINKED ISSUE #51: Allow users to get budget's by Id
Allow users to get a budget based on the id given.

This should return all details for the budget.

### PR DESCRIPTION
**Pull request by BenjaminGroseclose**
Closes #51

### CODE REVIEW

File: server/budgettracker.business/Api/BudgetApi.cs [hunk 1]
```diff
@@ -110,6 +111,27 @@ public async Task<ApiResponse> DeleteBudgets(ApiRequest request)
...
+        {
+            await Authenticate(request);
+
+            GetBudgetArgumentApiContract getBudget = request.Arguments<GetBudgetArgumentApiContract>();
+
+            ApiResponse response = new ApiResponse();
+
+            try
+            {
+                Budget retrievedBudget = await _budgetRepository.GetBudget(getBudget.BudgetValues.Id);
```
Comment (ianbro) [inline server/budgettracker.business/Api/BudgetApi.cs]: I think we should add a check in the API method to make sure the budget is actually the authenticated user's budget. That way they can't just ask for someone elses budget.

Comment (BenjaminGroseclose) [reply server/budgettracker.business/Api/BudgetApi.cs]: Acting like you ain't honeymooning. haha

Also can do ill add a userId check to confirm they belong to the authenticated user.

**Comment by BenjaminGroseclose** [comment]
@ianbro All good! Hope you are having fun!

File: server/budgettracker.business/Api/BudgetApi.cs [hunk 2]
```diff
@@ -113,15 +113,21 @@ public async Task<ApiResponse> DeleteBudgets(ApiRequest request)
...
             ApiResponse response = new ApiResponse();
 
             try
             {
-                Budget retrievedBudget = await _budgetRepository.GetBudget(getBudget.BudgetValues.Id);                
+                Budget retrievedBudget = await _budgetRepository.GetBudget(getBudget.BudgetValues.Id);
+
+                if (retrievedBudget.Owner.Id != user.Id) 
+                {
+                    throw new RepositoryException("Retrieve a budget with invalid user Id.");
```
Comment (ianbro) [inline server/budgettracker.business/Api/BudgetApi.cs]: When you are fixing Merge conflicts, consider changing this to something more official. Like `The authenticated user does not have permission to view this budget.`

What you have sounds like you're telling the user to retrieve the budget... which is what we're throwing an error about lol.

Comment (BenjaminGroseclose) [reply server/budgettracker.business/Api/BudgetApi.cs]: True true, will update

**Comment by BenjaminGroseclose** [comment]
@ianbro ready to go I think

**Pull request merged**